frontends/main/src/components/ErrorBoundary/ErrorBoundary.tsx

```diff
@@ -0,0 +1,81 @@
+"use client"
+
+import React, { Component } from "react"
+import NotFoundPage from "@/app-pages/ErrorPage/NotFoundPage"
+import ForbiddenPage from "@/app-pages/ErrorPage/ForbiddenPage"
+import { ForbiddenError } from "@/common/permissions"
+import { usePathname } from "next/navigation"
+
+interface ErrorBoundaryProps {
+  children: React.ReactNode
+}
+
+interface ErrorBoundaryHandlerProps extends ErrorBoundaryProps {
+  pathname: string
+}
+
+interface ErrorBoundaryHandlerState {
+  hasError: boolean
+  error: unknown
+  previousPathname: string
+}
+const isForbiddenError = (error: unknown) => error instanceof ForbiddenError
+
+class ErrorBoundaryHandler extends Component<
```

Comment:
I was interested to see a class component here and realize it's so we can use the error boundary lifecycle methods. It says here there's no equivalent for function components, https://legacy.reactjs.org/docs/hooks-faq.html#how-do-lifecycle-methods-correspond-to-hooks, though that doc is legacy... and these do appear in the up to date docs: https://react.dev/reference/react/Component#catching-rendering-errors-with-an-error-boundary (on the same page as a bunch of warnings "We recommend defining components as functions instead of classes")!

Looks like we can assume this is the recommended approach until they maybe make a hook available in future versions. The latest v19-beta does still recommend a single class component error boundary "if you’d like to avoid creating class components" - https://19.react.dev/reference/react/Component#componentdidcatch.